evennia/contrib/building_menu.py

```diff
@@ -0,0 +1,980 @@
+"""
+Module containing the building menu system.
+
+Evennia contributor: vincent-lg 2018
+
+Building menus are similar to `EvMenu`, except that they have been
+specifically designed to edit information as a builder.  Creating a
+building menu in a command allows builders quick-editing of a
+given object, like a room.  Here is an example of output you could
+obtain when editing the room:
+
+```
+ Editing the room: Limbo(#2)
+
+ [T]itle: the limbo room
+ [D]escription
+    This is the limbo room.  You can easily change this default description,
+    either by using the |y@desc/edit|n command, or simply by entering this
+    menu (enter |yd|n).
+ [E]xits:
+     north to A parking(#4)
+ [Q]uit this menu
+```
+
+From there, you can open the title choice by pressing t.  You can then
+change the room title by simply entering text, and go back to the
+main menu entering @ (all this is customizable).  Press q to quit this menu.
+
+The first thing to do is to create a new module and place a class
+inheriting from `BuildingMenu` in it.
+
+```python
+from evennia.contrib.building_menu import BuildingMenu
+
+class RoomBuildingMenu(BuildingMenu):
+    # ...
+```
+
+Next, override the `init` method.  You can add choices (like the title,
+description, and exits choices as seen above) by using the `add_choice`
+method.
+
+```
+class RoomBuildingMenu(BuildingMenu):
+    def init(self, room):
+        self.add_choice("title", "t", attr="key")
+```
+
+That will create the first choice, the title choice.  If one opens your menu
+and enter t, she will be in the title choice.  She can change the title
+(it will write in the room's `key` attribute) and then go back to the
+main menu using `@`.
+
+`add_choice` has a lot of arguments and offer a great deal of
+flexibility.  The most useful ones is probably the usage of callbacks,
+as you can set almost any argument in `add_choice` to be a callback, a
+function that you have defined above in your module.  This function will be
+called when the menu element is triggered.
+
+When you wish to create a building menu, you just need to import your
+class, create it specifying your intended caller and object to edit,
+then call `open`:
+
+```python
+from <wherever> import RoomBuildingMenu
+
+class CmdEdit(Command):
+
```

Comment:
This example needs a `key` for the Command. People will copy & paste this, you know! ;)